dashboard/app/models/script.rb

```diff
@@ -823,15 +827,15 @@ def get_script_level_by_id(script_level_id)
     script_levels.find(id: script_level_id.to_i)
   end
 
-  def get_script_level_by_relative_position_and_puzzle_position(relative_position, puzzle_position, lockable)
+  def get_script_level_by_relative_position_and_puzzle_position(relative_position, puzzle_position, survey)
     relative_position ||= 1
     script_levels.find do |sl|
       # make sure we are checking the native properties of the script level
       # first, so we only have to load lesson if it's actually necessary.
       sl.position == puzzle_position.to_i &&
         !sl.bonus &&
         sl.lesson.relative_position == relative_position.to_i &&
-        sl.lesson.lockable? == lockable
+        (survey ? sl.lesson.lockable? == true && sl.lesson.has_lesson_plan? == false : sl.lesson.lockable? == false || sl.lesson.has_lesson_plan? == true)
```

Comment:
nit: the `== true/false` parts of the clause aren't helpful to me and clutter this statement. This definitely could be personal preference though so if they make it clearer to others, feel free to ignore

Reply:
could some of this new logic be extracted into a variable with a descriptive name? I think this would be very hard for someone to understand without a lot of context.